### DISCUSSION TITLE
Local configuration file for backend config

### DISCUSSION THREAD

**Discussion by JayDoubleu** [pulumi/pulumi#13752] [General]
Is there a way to create a local configuration file that describes the backend configuration for a given folder, instead of using the global configuration file at ~/.pulumi?

I would like to be able to navigate from directory to directory without the two directories sharing their configuration or backend.

For example:

If I go to directory A and run pulumi up, it should use the s3 backend.
If I go to directory B and run pulumi up, it should use the azure-blob-storage backend.
If I check the contents of ~/.pulumi, it should be empty.

**Comment by ringods** [comment]
Hello @JayDoubleu, Pulumi doesn't have such a feature. 

However, in my personal setup I use [direnv](https://direnv.net/) which has the ability to load environment variables in your shell. It searches for the config starting from the local folder and traversing parent folders.

If you combine this by setting different values of `PULUMI_BACKEND_URL`, this should give you a setup matching your requirements. See the [Pulumi Environment Variables](https://www.pulumi.com/docs/cli/environment-variables/) for more info.
